src/Sulu/Bundle/AdminBundle/Resources/js/containers/Form/conditionDataProviders/parentConditionDataProvider.js

```diff
@@ -0,0 +1,10 @@
+// @flow
+import jsonpointer from 'json-pointer';
+
+export default function(data: Object, dataPath: ?string): {[string]: any} {
+    if (!dataPath) {
+        return {__parent: data};
+    }
+
+    return {__parent: jsonpointer.get(data, dataPath.substring(0, dataPath.lastIndexOf('/')))};
```

Comment:
I think it is not easily possible to implement this `__parent` variable recursively while still keeping the advantages of mobx's reactivity... I have tried something like this:

```javascript
// @flow
import {isObservableArray} from 'mobx';
import jsonpointer from 'json-pointer';

export default function(data: Object, dataPath: ?string): {[string]: any} {
    if (!dataPath) {
        return {__parent: data};
    }

    let parentDataPath = dataPath;
    const conditionData = {};
    let currentConditionData = conditionData;

    do {
        parentDataPath = parentDataPath.substring(0, parentDataPath.lastIndexOf('/'));
        const evaluatedData = jsonpointer.get(data, parentDataPath);

        if (Array.isArray(evaluatedData) || isObservableArray(evaluatedData)) {
            continue;
        }

        currentConditionData.__parent = evaluatedData;
        currentConditionData = currentConditionData.__parent;
    } while (parentDataPath.match(/^\/.*\//));

    return conditionData;
}
```

The problem is that the reference from the `ResourceStore` is manipulated here (adding the `__parent` variable). This ends in an endless loop, I guess that's because of some side effect... Even if we solve that issue, the side effects would still remain.

I guess using `toJS` somehow would work, but then too much code would be executed, because mobx cannot observe the single properties of that object correctly anymore, and making the rest a lot harder to implement. Also, I am not sure if that worse performance is a good tradeoff now, since we don't even know if somebody would make use of that...

Reply:
Maybe using a [Proxy](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy) would work :thinking: But then IE11 support is not possible anymore for sure...

Reply:
Nevermind, seems to work if the `evaluatedData` variable is spread. Also, still only the `render` method of the fields impacted by the currently edited field are called.

Reply:
you already solved this right? 🙂 
if not, maybe it would also work to evaluate the `__parent` lazy inside of a [getter on the object](https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Functions/get)?

Reply:
That wouldn't solve the problem, because we would still adjust the existing object. And yeah, I solved it by creating the objects using the `...` spread operator.